src/core/data.service.ts

```diff
@@ -43,18 +44,18 @@ export class DataService {
   /**
    * Notifies any data change listeners about a data change.
    *
-   * @param {ControlElement} uischema the affected control element
+   * @param {ControlElement} controlElement the affected control element
    * @param newValue the new value of the data chunk
    */
-  notifyChange(uischema: ControlElement, newValue: any): void {
-    if (uischema !== undefined && uischema !== null) {
-      const pair = getValuePropertyPair(this.data, uischema.scope.$ref);
+  notifyDataChangeListeners(controlElement: ControlElement, newValue: any): void {
```

Comment:
Hm, yes, true, I focused too much on the listener aspect here. Actually, we should probably rename and split this up to:

```JS
changeData(ControlElement, any) {
  // assign newValue as before
  notifyDataChangeListeners()
}
```

What do you think?